packages/polaris-viz/src/components/Grid/Grid.tsx

```diff
@@ -0,0 +1,699 @@
+        hoveredGroups={hoveredGroups}
+        handleGroupHover={handleGroupHover}
+        getColors={getColors}
+        containerWidth={dimensions.width}
+        isAnimated={isAnimated}
+      />
+    ));
+  };
+
+  const getColors = (group: CellGroup | null) => {
+    if (group) {
+      return {
+        bgColor: group.bgColor,
+        textColor: group.color,
+      };
+    }
+    return {
+      bgColor: 'blue',
+      textColor: '#000000',
+    };
+  };
+
+  const yTicks = useMemo(() => {
+    return Array.from({length: gridDimensions.rows}, (_, index) => ({
+      value: index,
+      label: `${index + 1}`,
+      formattedValue: `${index + 1}`,
+      yOffset: index * cellHeight + cellHeight / 2,
+    }));
+  }, [cellHeight, gridDimensions.rows]);
+
+  const xLabels = useMemo(
+    () =>
+      Array.from({length: gridDimensions.cols}, (_, index) => `${index + 1}`),
+    [gridDimensions.cols],
+  );
+
+  const xAxisLabelWidth = fullChartWidth / xLabels.length;
+
+  const xScale = useMemo(
+    () =>
+      scaleLinear().domain([0, gridDimensions.cols]).range([0, fullChartWidth]),
+    [gridDimensions.cols, fullChartWidth],
+  );
+
+  const renderTooltip = () => {
+    if (!tooltipInfo) return null;
+
+    const {x, y, groupName, groupDescription, groupGoal} = tooltipInfo;
+    const transform = `translate(${x}, ${y})`;
+
+    return (
+      <g transform={transform}>
+        <foreignObject x={0} y={0} width={TOOLTIP_WIDTH} height={tooltipHeight}>
+          <div
+            ref={(node) => {
+              if (node) {
+                const {height} = node.getBoundingClientRect();
+                const tooltipHeight = height + TOOLTIP_PADDING * 2;
+                setTooltipHeight(tooltipHeight);
+              }
+            }}
+            className={styles.Tooltip}
+            data-testid="tooltip"
+          >
+            <div className={styles.TooltipTitle}>{groupName}</div>
+            <div className={styles.TooltipDescription}>{groupDescription}</div>
+
+            {groupGoal && (
+              <div className={styles.TooltipGoal}>
+                <svg
+                  xmlns="http://www.w3.org/2000/svg"
+                  fill="none"
+                  viewBox="0 0 24 24"
+                  strokeWidth="2.5"
+                  stroke="currentColor"
+                  style={{height: '13px', width: '20px'}}
+                >
+                  <path
+                    strokeLinecap="round"
+                    strokeLinejoin="round"
+                    d="M13.5 4.5 21 12m0 0-7.5 7.5M21 12H3"
+                  />
+                </svg>
+
+                <p className={styles.GroupGoal}>{groupGoal}</p>
+              </div>
+            )}
+          </div>
+        </foreignObject>
+      </g>
+    );
+  };
+
+  useEffect(() => {
+    // trigger animation after a short delay to ensure DOM is ready
+    const timer = setTimeout(() => {
+      setAnimationStarted(true);
+    }, 100);
```

Comment:
i'm kind of confused why this tracking of the `animationStarted` is needed, aren't you animating things with CSS only or am i missing something? 🤔